src/kv.c

```diff
@@ -1178,16 +1181,31 @@ kvStore_WatchAll(kvWatcher **new_watcher, kvStore *kv, kvWatchOptions *opts)
 
 natsStatus
 kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts)
+{
+    if (list == NULL)
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    return _kvStore_Keys(list, kv, opts, NULL, 0);
+}
+
+natsStatus
+kvStore_KeysWithFilters(kvKeysList *list, kvStore *kv, kvWatchOptions *opts, const char **filters, int numFilters) {
+
+    if (list == NULL || filters == NULL || numFilters == 0)
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    return _kvStore_Keys(list, kv, opts, filters, numFilters);
+}
+
+static natsStatus
+_kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts, const char **filters, int numFilters)
 {
     natsStatus      s;
     kvWatchOptions  o;
     kvWatcher       *w = NULL;
     int             count = 0;
     KV_DEFINE_LIST;
 
-    if (list == NULL)
```

Comment:
Would have left this one here since it is common to both Keys() and KeysWithFilter()